Edge api

evirens opened the pull request:
Edge API created with models, controllers. 
NOTE: not yet on (I think) dockerhub, where the okteto deploy looks for it?

Commit:
Merge branch 'dev' into Ebert_Dev

Commit:
framework edge api
includes models, no logic

Commit:
Added controller for instrument
not gonna work due to not setting application properties, that's the next step or something

Commit:
Environment

Added dockerfile, application properties and the first controller method

Commit:
Merge branch 'dev' into Ebert_Dev

Commit:
All controllers and mappings

Commit:
wording :)

Dzetacq commented:
wow, sonarcloud really be spamming here

JohanTorfs commented on .github/workflows/all_docker_deploy.yml:
```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       matrix:
-        api-name: [instrument_api, piece_api, review_api]
+        api-name: [instrument_api, piece_api, review_api, edge_api]
```
This should work.

JohanTorfs commented on edge_api/Dockerfile:
```diff
@@ -0,0 +1,5 @@
+EXPOSE 8050
+ARG JAR_FILE=target/*.jar
+COPY ${JAR_FILE} app.jar
+ENTRYPOINT ["java","-jar","/app.jar"]
```
You need to add the environment variables (from the application properties file) here. This is needed so the container can find the other containers in the kubernetes cluster

JohanTorfs commented on edge_api/pom.xml:
```diff
@@ -0,0 +1,56 @@
+    <name>edge_api</name>
+    <description>edge_api</description>
+    <properties>
+        <java.version>17</java.version>
```
Add the sonarcloud properties. Although they will mostly be overwritten by the input from the workflows. (Look at the other microservices)

JohanTorfs commented on edge_api/pom.xml:
```diff
@@ -0,0 +1,56 @@
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.springframework.boot</groupId>
```
There is an other plugin that needs to be added for sonarcloud to do the coverage checks. Forgot the name of the plugin, but you can find it in the other microservice files

JohanTorfs commented on edge_api/src/main/java/com/example/edge_api/controller/InstrumentController.java:
```diff
@@ -0,0 +1,44 @@
+    private String pieceServiceBaseUrl;
+
+    @GetMapping("/instrument/{name}")
+    public Instrument getInstrumentWithMusiciansAndPieces(@PathVariable String name) {
```
Does this not need to return a list? It just seems more logical that it would return a list of instruments that match the given string.

JohanTorfs replied:
Uitstellen tot front end

JohanTorfs commented on edge_api/src/main/java/com/example/edge_api/controller/MusicianController.java:
```diff
@@ -0,0 +1,39 @@
+    private String instrumentServiceBaseUrl;
+
+    @GetMapping("/musician/{name}")
+    public Musician getMusicianWithInstrument(@PathVariable String name) {
```
Same here. Is a list not more usable?

JohanTorfs replied:
Front end

JohanTorfs commented on edge_api/src/main/resources/application.properties:
```diff
@@ -0,0 +1,5 @@
+server.port=8050
+
+instrumentservice.baseurl= ${INSTRUMENT_SERVICE_BASEURL:localhost:8051}
```
Ik denk dat de ports niet gaan werken op okteto, dus hier zou je best ook een env variabele toevoegen. Dit moeten we misschien eerst even in een lokale cluster testen.

JohanTorfs commented on okteto-compose.yml:
```diff
@@ -66,6 +66,18 @@ services:
+      environment:
+        INSTRUMENT_SERVICE_BASEURL: instrument-service:8051
+        PIECE_SERVICE_BASEURL: piece-service:8052
+        REVIEW_SERVICE_BASEURL: review-service:8053
```
Waarschijnlijk goed als we deze environment variabelen ook in de dockerfile zetten. De push naar docker en okteto gebeurd trouwens enkel op push naar main.

JohanTorfs replied:
Ik denk trouwens ook dat de ports hier wegmogen

JohanTorfs reviewed (changes requested):
Ziet er heel goed uit, heb een paar opmerkingen en een kleine aanpassing voor de dockerfile. Eens dat dit klaar is, zal ik alles eerst eens lokaal proberen testen.

JohanTorfs committed:
Merge branch 'main' of github.com:JohanTorfs/instrument_edge_api into Ebert_Dev

JohanTorfs committed:
fixes to edge api

Commit:
Comments

Commit:
Merge branch 'Ebert_Dev' of https://github.com/JohanTorfs/instrument_edge_api into Ebert_Dev

Commit:
Merge branch 'dev' into Ebert_Dev

Commit:
Fixed review & edge api

All endpoints excluding musicians and instruments fixed
some changes needed to review api, which now only stores piece names and not their ID, it was a strange combination of both before, making for double data storage

Johan-T42 committed:
fix application properties
